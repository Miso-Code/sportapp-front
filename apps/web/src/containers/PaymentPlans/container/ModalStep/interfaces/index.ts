import { FormPaymentData } from '@/containers/PaymentForm/utils/schema'

export interface Props {
	open: boolean
	handleSubmit: (data: FormPaymentData) => void
	handleClose: () => void,
	isFreePlan: boolean
	price: number
	isLoading: boolean
}
