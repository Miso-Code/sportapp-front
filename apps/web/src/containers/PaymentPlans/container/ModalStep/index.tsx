import PaymentForm from '@/containers/PaymentForm'
import { FormPaymentData } from '@/containers/PaymentForm/utils/schema'
import LoadingButton from '@mui/lab/LoadingButton'
import {
	Box,
	Button,
	CardActions,
	CardContent,
	Modal,
	Typography
} from '@mui/material'
import { useTranslation } from 'react-i18next'
import { Props } from './interfaces'

const style = {
	position: 'absolute',
	top: '50%',
	left: '50%',
	transform: 'translate(-50%, -50%)',
	minWidth: 300,
	width: '100%',
	maxWidth: 800,
	bgcolor: 'background.paper',
	border: '2px solid #000',
	boxShadow: 24,
	p: 4
}

export default function ModalStep({
	handleClose,
	handleSubmit,
	open,
	price,
	isFreePlan,
	isLoading
}: Props) {
	const { t } = useTranslation()

	const handleFreeSubmit = () => {
		const payload: FormPaymentData = {
			amount: 0,
			cvc: '123',
			expiry: '12/90',
			name: 'Test User',
			number: '4242424242424242'
		}
		handleSubmit(payload)
	}

	return (
		<>
			<Modal
				open={open}
				onClose={handleClose}
				aria-labelledby='modal-modal-title'
				aria-describedby='modal-modal-description'>
				<Box sx={style}>
					{isFreePlan ? (
						<Box>
							<CardContent>
								<Typography
									gutterBottom
									variant='h5'
									component='div'>
									{t('paymentPlans.freeCard.title')}
								</Typography>
								<Typography
									variant='body2'
									color='text.secondary'>
									{t('paymentPlans.freeCard.description')}
								</Typography>
							</CardContent>
							<CardActions>
								<Button
									onClick={handleClose}
									color='error'
									fullWidth
									variant='contained'>
									{t('form.cancel')}
								</Button>
								<LoadingButton
									color='success'
									fullWidth
									onClick={handleFreeSubmit}
									loading={isLoading}
									variant='contained'>
									{t('form.obtain')}
								</LoadingButton>
							</CardActions>
						</Box>
					) : (
						<PaymentForm
							onCancel={handleClose}
							onSubmit={handleSubmit}
							options={[
								{
									label: `$${price}`,
									value: `${price}`
								}
							]}
							price={price}
							isLoading={isLoading}
						/>
					)}
				</Box>
			</Modal>
		</>
	)
}
