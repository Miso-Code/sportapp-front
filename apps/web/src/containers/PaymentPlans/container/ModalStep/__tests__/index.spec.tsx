import ModalStep from '@/containers/PaymentPlans/container/ModalStep'
import {
	act,
	fireEvent,
	render,
	RenderResult,
	waitFor
} from '@testing-library/react'

describe('ModalStep', () => {
	let wrapper: RenderResult

	beforeEach(() => {
		wrapper = render(
			<ModalStep
				handleClose={jest.fn()}
				handleSubmit={jest.fn()}
				open={true}
				isFreePlan={false}
				isLoading={false}
				price={100}
			/>
		)
	})

	it('should render the component', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it('should render the component with free plan', () => {
		wrapper.rerender(
			<ModalStep
				handleClose={jest.fn()}
				handleSubmit={jest.fn()}
				open={true}
				isFreePlan={true}
				isLoading={false}
				price={0}
			/>
		)

		expect(wrapper).toMatchSnapshot()
	})

	it('should call handleSubmit when free plan is selected', async () => {
		const handleSubmit = jest.fn()
		wrapper.rerender(
			<ModalStep
				handleClose={jest.fn()}
				handleSubmit={handleSubmit}
				open={true}
				isFreePlan={true}
				isLoading={false}
				price={0}
			/>
		)

		act(() => {
			fireEvent.click(wrapper.getByText('form.obtain'))
		})

		await waitFor(() => {
			expect(handleSubmit).toHaveBeenCalledWith({
				amount: 0,
				cvc: '123',
				expiry: '12/90',
				name: 'Test User',
				number: '4242424242424242'
			})
		})
	})
})
