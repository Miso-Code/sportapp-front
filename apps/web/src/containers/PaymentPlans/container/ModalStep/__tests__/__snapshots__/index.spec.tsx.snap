// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalStep should render the component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      aria-describedby="modal-modal-description"
      aria-labelledby="modal-modal-title"
      class="MuiModal-root css-79ws1d-MuiModal-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiBox-root css-15fsfc9"
        tabindex="-1"
      >
        <form>
          <div
            class="rccs"
          >
            <div
              class="rccs__card rccs__card--unknown"
            >
              <div
                class="rccs__card--front"
              >
                <div
                  class="rccs__card__background"
                />
                <div
                  class="rccs__issuer"
                />
                <div
                  class="rccs__cvc__front"
                />
                <div
                  class="rccs__number"
                >
                  •••• •••• •••• ••••
                </div>
                <div
                  class="rccs__name rccs--focused"
                >
                  form.card.cardHolderPlaceholder
                </div>
                <div
                  class="rccs__expiry"
                >
                  <div
                    class="rccs__expiry__valid"
                  >
                    valid thru
                  </div>
                  <div
                    class="rccs__expiry__value"
                  >
                    ••/••
                  </div>
                </div>
                <div
                  class="rccs__chip"
                />
              </div>
              <div
                class="rccs__card--back"
              >
                <div
                  class="rccs__card__background"
                />
                <div
                  class="rccs__stripe"
                />
                <div
                  class="rccs__signature"
                />
                <div
                  class="rccs__cvc"
                />
                <div
                  class="rccs__issuer"
                />
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-1oke1pi"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                id="amount-select-label"
              >
                form.amount
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl  css-1ualgfl-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              >
                <div
                  aria-controls=":r0:"
                  aria-disabled="true"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="amount-select-label mui-component-select-amount"
                  class="MuiSelect-select MuiSelect-outlined Mui-disabled MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-component-select-amount"
                  role="combobox"
                >
                  $100
                </div>
                <input
                  aria-hidden="true"
                  aria-invalid="false"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  disabled=""
                  id="amount"
                  name="amount"
                  tabindex="-1"
                  value="100"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined Mui-disabled css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-14lo706"
                  >
                    <span>
                      form.amount
                    </span>
                  </legend>
                </fieldset>
              </div>
              <p
                class="MuiFormHelperText-root Mui-disabled MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
              />
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r1:"
                id=":r1:-label"
              >
                form.card.cardNumber
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="number"
                  name="number"
                  type="number"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      form.card.cardNumber
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r2:"
                id=":r2:-label"
              >
                form.card.cardHolder
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="name"
                  name="name"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      form.card.cardHolder
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiBox-root css-p27s4m"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1g1iitx-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for=":r3:"
                  id=":r3:-label"
                >
                  form.card.expirationDate
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="expiry"
                    name="expiry"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        form.card.expirationDate
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiTextField-root css-1g1iitx-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for=":r4:"
                  id=":r4:-label"
                >
                  form.card.cvv
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="cvc"
                    name="cvc"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        form.card.cvv
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-ke9lij"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorError MuiButton-fullWidth MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorError MuiButton-fullWidth css-1e0debn-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
                id=":r5:"
                tabindex="0"
                type="button"
              >
                form.cancel
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorSuccess MuiButton-fullWidth Mui-disabled MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorSuccess MuiButton-fullWidth css-xtcf48-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
                disabled=""
                id=":r6:"
                tabindex="-1"
                type="submit"
              >
                form.checkout
              </button>
            </div>
          </div>
        </form>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ModalStep should render the component with free plan 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      aria-describedby="modal-modal-description"
      aria-labelledby="modal-modal-title"
      class="MuiModal-root css-79ws1d-MuiModal-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiBox-root css-15fsfc9"
        tabindex="-1"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
            >
              paymentPlans.freeCard.title
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              paymentPlans.freeCard.description
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-dnrpxu-MuiCardActions-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError MuiButton-fullWidth css-yup6gp-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              form.cancel
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess MuiButton-fullWidth MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess MuiButton-fullWidth css-zip927-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
              id=":re:"
              tabindex="0"
              type="button"
            >
              form.obtain
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
